pool/paymentmgr.go

```diff
@@ -904,53 +916,64 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32) error {
 		outs[addr] = amt
 	}
 
-	tx, err := txCreator.CreateRawTransaction(ctx, inputs, outs, nil, nil)
-	if err != nil {
-		return fmt.Errorf("unable to create raw transaction: %v", err)
+	// Ensure the wallet is aware of all the outputs to be spent by the payout
+	// transaction.
+	if maxSpendableHeight < height {
+		maxSpendableHeight = height
 	}
 
-	txB, err := tx.Bytes()
+	hashes := make([]*chainhash.Hash, 0, len(txHashes))
+	for _, hash := range txHashes {
+		hashes = append(hashes, hash)
+	}
+
+	notifSource, err := pm.cfg.GetTxConfNotifications(hashes,
+		int32(maxSpendableHeight))
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to stream tx confirmations: %v", err)
 	}
 
-	// TODO: With dcrwallet notifications on the coinbase tx like jholdstock
-	// suggested we would not need to be watching the wallet like this or even
-	// make balance queries.
+	// Wait for coinbase tx confirmations from the wallet.
+	maxSpendableConfs := int32(pm.cfg.ActiveNet.CoinbaseMaturity) + 1
 	for {
-		// Ensure the wallet is synced to the current chain height before
-		// proceeding with payments.
-		walletHeight, err := pm.cfg.WalletBestHeight(ctx)
+		resp, err := notifSource()
 		if err != nil {
-			return fmt.Errorf("unable to fetch best wallet height: %v", err)
+			return fmt.Errorf("tx confirmations notification error: %v", err)
+		}
+
+		// Ensure all coinbases being spent are spendable before proceeding
+		// with creating and publishing the transaction.
+		for _, coinbase := range resp.Confirmations {
+			if coinbase.Confirmations >= maxSpendableConfs {
+				cbHash, err := chainhash.NewHash(coinbase.TxHash)
+				if err != nil {
+					return fmt.Errorf("unable to create block hash: %v", err)
+				}
+
+				// Remove spendable coinbases from the tx hash set. All
+				// coinbases are spendable when the tx hash set is empty.
+				delete(txHashes, cbHash.String())
+			}
 		}
 
-		if walletHeight >= height {
-			time.Sleep(time.Second * 3)
+		if len(txHashes) == 0 {
```

Comment:
Should we close notification stream here?

Reply:
Nope, the stream is terminated at `maxSpendableHeight`.